src/github.com/getlantern/errlog/errlog.go

```diff
@@ -0,0 +1,529 @@
+/*
+Package errlog defines error types used across Lantern project and implements
+functions to manipulate them.
+
+  var elog = errlog.ErrorLoggerFor("package-name")
+  if n, err := Foo(); err != nil {
+    elog.Log(err, errlog.WithOp("foo"), errlog.WithField("foo": "bar"))
+  }
+
```

Comment:
@fffw You could follow the same convention as the built-in log package, which accepts an interface{} and then treats the value appropriately.  Regarding optionality - the conventional thing in Go is to leave the fields that you're not interested out of your struct initializer and the receiving function then fills those in.  I'll do a little PR to show what I'm suggesting.